.github/workflows/quality.yml

```diff
@@ -5,26 +5,9 @@ on: [pull_request]
 jobs:
   check_code_quality:
     runs-on: ubuntu-latest
-    env:
-      UV_HTTP_TIMEOUT: 600 # max 10min to install deps
-
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
-          python-version: "3.12"
-
-      # Setup venv
-      - name: Setup venv + uv
```

Comment:
pre-commit manages its own isolated venv, hence no more env management.
Also no need to install `smolagents` for lint & formating, only `pre-commit` is needed